internal/peermanager/peerwantmanager.go

```diff
@@ -28,39 +40,45 @@ type peerWantManager struct {
 	wantPeers map[cid.Cid]map[peer.ID]struct{}
 
 	// broadcastWants tracks all the current broadcast wants.
-	broadcastWants *cid.Set
+	broadcastWants cidSessSet
 
 	// Keeps track of the number of active want-blocks
 	wantBlockGauge Gauge
 }
 
 type peerWant struct {
-	wantBlocks *cid.Set
-	wantHaves  *cid.Set
+	wants     map[cid.Cid]*wantInfo
+	peerQueue PeerQueue
 }
```

Comment:
The want is either a want-have or a want-block, so unify those two lists and add a flag indicating the want type: wantInfo.tp